kernel/src/snapshot.rs

```diff
@@ -297,7 +302,9 @@ fn read_last_checkpoint(
         .read_files(vec![(file_path, None)])
         .and_then(|mut data| data.next().expect("read_files should return one file"))
     {
-        Ok(data) => Ok(Some(serde_json::from_slice(&data)?)),
+        Ok(data) => Ok(serde_json::from_slice(&data)
+            .map_err(|e| warn!("invalid _last_checkpoint JSON: {e}"))
```

Comment:
You probably want `inspect_err` instead https://doc.rust-lang.org/std/result/enum.Result.html#method.inspect_err